Subject: [PATCH] Carry SDK37 public type splits into the original API30 mapping

Use the same public aliases in the target/current API31 mapping as the
reference for additive SDK37 type splits. Preserve every original API30
member, declaration and compat entry; do not relabel an API31 mapping.

diff --git a/mapping/30.0.cil b/mapping/30.0.cil
--- a/mapping/30.0.cil
+++ b/mapping/30.0.cil
@@ -1778,12 +1778,12 @@
 (typeattributeset performanced_exec_30_0 (performanced_exec))
 (typeattributeset permission_service_30_0 (permission_service))
 (typeattributeset permissionmgr_service_30_0 (permissionmgr_service))
-(typeattributeset persist_debug_prop_30_0 (persist_debug_prop))
+(typeattributeset persist_debug_prop_30_0 (persist_debug_prop system_user_mode_emulation_prop))
 (typeattributeset persistent_data_block_service_30_0 (persistent_data_block_service))
 (typeattributeset persistent_properties_ready_prop_30_0 (persistent_properties_ready_prop))
 (typeattributeset pinner_service_30_0 (pinner_service))
 (typeattributeset pipefs_30_0 (pipefs))
-(typeattributeset platform_app_30_0 (platform_app))
+(typeattributeset platform_app_30_0 (platform_app platform_app_36))
 (typeattributeset platform_compat_service_30_0 (platform_compat_service))
 (typeattributeset pm_prop_30_0 (pm_prop))
 (typeattributeset pmsg_device_30_0 (pmsg_device))
@@ -1802,9 +1802,9 @@
 (typeattributeset preloads_media_file_30_0 (preloads_media_file))
 (typeattributeset prereboot_data_file_30_0 (prereboot_data_file))
 (typeattributeset print_service_30_0 (print_service))
-(typeattributeset priv_app_30_0 (priv_app))
+(typeattributeset priv_app_30_0 (priv_app priv_app_36))
 (typeattributeset privapp_data_file_30_0 (privapp_data_file))
-(typeattributeset proc_30_0 (proc proc_bootconfig proc_bpf proc_cpu_alignment proc_watermark_scale_factor))
+(typeattributeset proc_30_0 (proc proc_bootconfig proc_bpf proc_cpu_alignment proc_watermark_scale_factor proc_firmware_config))
 (typeattributeset proc_abi_30_0 (proc_abi))
 (typeattributeset proc_asound_30_0 (proc_asound))
 (typeattributeset proc_bluetooth_writable_30_0 (proc_bluetooth_writable))
@@ -1816,7 +1816,7 @@
 (typeattributeset proc_drop_caches_30_0 (proc_drop_caches))
 (typeattributeset proc_extra_free_kbytes_30_0 (proc_extra_free_kbytes))
 (typeattributeset proc_filesystems_30_0 (proc_filesystems))
-(typeattributeset proc_fs_verity_30_0 (proc_fs_verity))
+(typeattributeset proc_fs_verity_30_0 (proc_fs_verity proc proc_firmware_config))
 (typeattributeset proc_hostname_30_0 (proc_hostname))
 (typeattributeset proc_hung_task_30_0 (proc_hung_task))
 (typeattributeset proc_interrupts_30_0 (proc_interrupts))
@@ -1995,7 +1995,7 @@
 (typeattributeset surfaceflinger_service_30_0 (surfaceflinger_service))
 (typeattributeset surfaceflinger_tmpfs_30_0 (surfaceflinger_tmpfs))
 (typeattributeset swap_block_device_30_0 (swap_block_device))
-(typeattributeset sysfs_30_0 (sysfs sysfs_fs_incfs_features sysfs_fs_incfs_metrics))
+(typeattributeset sysfs_30_0 (sysfs sysfs_fs_incfs_features sysfs_fs_incfs_metrics sysfs_block sysfs_mem_sleep sysfs_powercap sysfs_thunderbolt sysfs_typec sysfs_udc sysfs_usb_pd sysfs_usbcore))
 (typeattributeset sysfs_android_usb_30_0 (sysfs_android_usb))
 (typeattributeset sysfs_batteryinfo_30_0 (sysfs_batteryinfo))
 (typeattributeset sysfs_bluetooth_writable_30_0 (sysfs_bluetooth_writable))
@@ -2131,7 +2131,7 @@
 (typeattributeset untrusted_app_25_30_0 (untrusted_app_25))
 (typeattributeset untrusted_app_27_30_0 (untrusted_app_27))
 (typeattributeset untrusted_app_29_30_0 (untrusted_app_29))
-(typeattributeset untrusted_app_30_0 (untrusted_app untrusted_app_32))
+(typeattributeset untrusted_app_30_0 (untrusted_app untrusted_app_32 untrusted_app_34))
 (typeattributeset update_engine_30_0 (update_engine))
 (typeattributeset update_engine_data_file_30_0 (update_engine_data_file))
 (typeattributeset update_engine_exec_30_0 (update_engine_exec))
@@ -2167,7 +2167,7 @@
 (typeattributeset vendor_configs_file_30_0 (vendor_apex_metadata_file vendor_configs_file))
 (typeattributeset vendor_data_file_30_0 (vendor_data_file vendor_userdir_file))
 (typeattributeset vendor_default_prop_30_0 (vendor_default_prop))
-(typeattributeset vendor_file_30_0 (vendor_file))
+(typeattributeset vendor_file_30_0 (vendor_file vendor_boot_ota_file))
 (typeattributeset vendor_framework_file_30_0 (vendor_framework_file))
 (typeattributeset vendor_hal_file_30_0 (vendor_hal_file))
 (typeattributeset vendor_idc_file_30_0 (vendor_idc_file))
